packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/dialogs/duplicateobject/services/DuplicateObjectDialogTreeDescriptionProvider.java

```diff
@@ -0,0 +1,339 @@
+import org.springframework.stereotype.Service;
+
+/**
+ * This class is used to provide the description of the model browser tree.
+ *
+ * @author pcdavid
+ */
+@Service
+public class DuplicateObjectDialogTreeDescriptionProvider implements IEditingContextRepresentationDescriptionProvider {
+
+    public static final String DESCRIPTION_ID = UUID.nameUUIDFromBytes("duplicate_object_dialog_tree_description".getBytes()).toString();
+
+    public static final String REPRESENTATION_NAME = "Duplicate Object Dialog Tree";
+
+    public static final String DOCUMENT_KIND = "siriusWeb://document";
+
+    public static final String PREFIX = "duplicateObjectDialogTree://";
+
+    private final IObjectService objectService;
+
+    private final IURLParser urlParser;
+
+    private final IEMFKindService emfKindService;
+
+
+    public DuplicateObjectDialogTreeDescriptionProvider(IObjectService objectService, IURLParser urlParser, IEMFKindService emfKindService) {
+        this.objectService = Objects.requireNonNull(objectService);
+        this.urlParser = Objects.requireNonNull(urlParser);
+        this.emfKindService = Objects.requireNonNull(emfKindService);
+    }
+
+    @Override
+    public List<IRepresentationDescription> getRepresentationDescriptions(IEditingContext editingContext) {
+
+        Function<VariableManager, Boolean> isSelectableProvider = variableManager -> {
+            EClass referenceKind = this.resolveReferenceEClass(variableManager).orElse(null);
+            return this.isContainerSelectable(variableManager, referenceKind);
+        };
+        return List.of(TreeDescription.newTreeDescription(DuplicateObjectDialogTreeDescriptionProvider.DESCRIPTION_ID)
+                .label(REPRESENTATION_NAME)
+                .idProvider(variableManager -> variableManager.get(GetOrCreateRandomIdProvider.PREVIOUS_REPRESENTATION_ID, String.class).orElse("duplicateObjectDialogTree://"))
+                .treeItemIdProvider(this::getTreeItemId)
+                .kindProvider(this::getKind)
+                .labelProvider(this::getLabel)
+                .targetObjectIdProvider(variableManager -> variableManager.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class)
+                        .map(IEditingContext::getId)
+                        .orElse(null))
+                .treeItemIconURLsProvider(this::getImageURL)
+                .editableProvider(this::isEditable)
+                .deletableProvider(this::isDeletable)
+                .selectableProvider(isSelectableProvider)
+                .elementsProvider(this::getCreationScopeElements)
+                .hasChildrenProvider(this::hasChildren)
+                .childrenProvider(this::getChildren)
+                // This predicate will NOT be used while creating the model browser, but we don't want to see the description of the
+                // model browser in the list of representations that can be created. Thus, we will return false all the time.
```

Comment:
Is there a point in duplicating the model browser? Why pick this approach. As I said, I would have expected a first PR to extract the model browser instead of just copying and pasting half a dozen files. Unless there is a point that I'm missing?